src/widgets/table.rs

```diff
@@ -401,10 +401,15 @@ impl<'a> Table<'a> {
     fn get_columns_widths(&self, max_width: u16, selection_width: u16) -> Vec<(u16, u16)> {
         // Default to equal column widths if not provided
         let widths = if self.widths.is_empty() {
-            let max_col_count = self.rows.iter().map(|r| r.cells.len()).max();
-            let header_col_count = self.header.as_ref().unwrap_or(&Row::default()).cells.len();
-            let col_count = max_col_count.unwrap_or(header_col_count);
-            vec![Constraint::Percentage(100 / col_count as u16); col_count]
+            let max_col_count = self
+                .rows
+                .iter()
+                .chain(self.header.iter())
+                .map(|r| r.cells.len())
+                .max()
+                .unwrap_or(0);
+            let equal_length = max_width / max_col_count as u16 - 2 * self.column_spacing;
+            vec![Constraint::Length(equal_length); max_col_count]
```

Comment:
Can you add some documentation that explains why this calculation works? Seems magical to subtract 2*column_spacing.

Reply:
Definitely, though I'm just realising that it's actually incorrect. My thinking was that each column should share an equal portion of the `max_width`, minus the space that the column spacing takes up. The formula is wrong, though. I'll correct it and make it clearer.